ipatests/test_integration/test_sssd.py

```diff
@@ -403,3 +405,141 @@ def test_trustdomain_disable_disables_subdomain(self):
                 assert 'no such user' in res.stderr_text
         # verify the user can be retrieved after re-enabling trustdomain
         self.master.run_command(['id', user])
+
+
+class TestNestedMembers(IntegrationTest):
+    num_clients = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(cls.master)
+        tasks.install_client(cls.master, cls.clients[0])
+
+    def test_nested_group_members(self):
+        """Nested group memberships should be honoured
+
+        "c" should be a child of "b"
+        so that parent child relationship is as follows:
+        "a"->"b"->"c"
+
+        user1 is direct member of "a" and as a result
+        member of "b" and "c"".
+        Now if one adds a direct membership to "b"
+        nothing will change.
+
+        Now if one removes the direct membership to "b"
+        nothing should change, the memberships should be honored
+        Linked Issue: https://pagure.io/SSSD/sssd/issue/3636
+        """
+        master = self.master
+        client = self.clients[0]
+        tasks.kinit_admin(master)
+
+        # add hbacrule
+        master.run_command(['ipa', 'hbacrule-add', '--usercat', 'all',
+                            '--hostcat', 'all', '--servicecat', 'all',
+                            'any_to_any'])
+        # add a user
+        username = "testuser" + str(random.randint(200000, 9999999))
```

Comment:
As we are not using more user here, `testuser001` or something like that would be enough.
Make sure to remove the user/group at the end of test as a cleanup task.

Reply:
Fixed it